src/endpoints/authentication/jwt_auth.cpp

```diff
@@ -153,13 +152,14 @@ namespace ccf
       }
     }
 
-    if (!token_keys)
+    if (!token_keys || token_keys->empty())
     {
       error_reason =
         fmt::format("JWT signing key not found for kid {}", key_id);
       return nullptr;
     }
 
+    error_reason = "Cant authenticate JWT token: no matching public key found";
```

Comment:
```suggestion
    error_reason = "Cannot authenticate JWT token: no matching public key found";
```

Reply:
This is not needed, instead line 167 should set error_reason to something like "Signature verification failed".

Reply:
Done